asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py

```diff
@@ -0,0 +1,850 @@
+    save_openeye_sdf,
+    split_openeye_design_unit,
+)
+from asapdiscovery.data.schema import CrystalCompoundData, ExperimentalCompoundData
+from asapdiscovery.data.utils import (
+    exp_data_to_oe_mols,
+    is_valid_smiles,
+    oe_load_exp_from_file,
+)
+from asapdiscovery.dataviz.gif_viz import GIFVisualizer
+from asapdiscovery.dataviz.html_viz import HTMLVisualizer
+from asapdiscovery.docking import dock_and_score_pose_oe, make_docking_result_dataframe
+from asapdiscovery.docking import prep_mp as oe_prep_function
+from asapdiscovery.simulation.simulate import VanillaMDSimulator
+
+"""
+Script to run single target prep + docking.
+
+Input:
+    - receptor: path to receptor to prep and dock to
+    - mols: path to the molecules to dock to the receptor as an SDF or SMILES file, or SMILES string.
+    - title: title of molecule to use if a SMILES string is passed in as input, default is to use an index
+    - smiles_as_title: use smiles strings as titles for molecules in .smi or .sdf file if none provided
+    - dask: use dask to parallelise docking
+    - dask-lilac: run dask in lilac config mode
+    - output_dir: path to output_dir, will NOT overwrite if exists.
+    - debug: enable debug mode, with more files saved and more verbose logging
+    - verbose: whether to print out verbose logging.
+    - cleanup: clean up intermediate files except for final csv and visualizations
+    - logname: name of logger
+
+    # Prep arguments
+    - loop_db: path to loop database.
+    - seqres_yaml: path to yaml file of SEQRES.
+    - protein_only: if true, generate design units with only the protein in them
+    - ref_prot: path to reference pdb to align to. If None, no alignment will be performed
+
+    # Docking arguments
+    - docking_sys: which docking system to use [posit, hybrid]. Defaults to posit.
+    - relax: when to run relaxation [none, clash, all]. Defaults to none.
+    - omega: use Omega conformer enumeration.
+    - hybrid: whether to only use hybrid docking protocol in POSIT.
+    - num_poses: number of poses to return from docking.
+    - gat: whether to use GAT model to score docked poses.
+
+
+    # MD arguments
+    - md: whether to run MD after docking.
+    - md-steps: number of MD steps to run, default 2500000 for a 10 ns simulation at 4 fs timestep.
+
+Example usage:
+
+    # with an SDF or SMILES file
+
+    dock-small-scale-e2e \
+        -r /path/to/receptor.pdb \
+        -m /path/to/mols.[sdf/smi] \
+        -o /path/to/output_dir \
+
+    # with a SMILES string using dask and requesting MD
+
+    dock-small-scale-e2e \
+        -r /path/to/receptor.pdb \
+        -m 'COC(=O)COc1cc(cc2c1OCC[C@H]2C(=O)Nc3cncc4c3cccc4)Cl'
+        -o /path/to/output_dir \
+        --title 'my_fancy_molecule' \\ # without a title you will be given a hash of the SMILES string
+        --debug \
+        --dask \
+        --md
+
+"""
+
+
+# setup input arguments
+parser = argparse.ArgumentParser(description="Run single target docking.")
```

Comment:
@apayne97 @kaminow would you be able to check that the defaults for docking and  prep in the CLI are the most rigorous.

Reply:
```
parser.add_argument(
    "--relax",
    default="none",
    help="When to run relaxation [none, clash, all]. Defaults to none.",
)

parser.add_argument(
    "--omega",
    action="store_true",
    help="Use Omega conformer enumeration.",
)
```
These would be the main things to make it more rigorous - I think the best would be to have `relax=clash` and omega set to true by default. However it does make the docking take ~10x longer. So I think it would be recommended that it should be that way by default but for large docking jobs then you might want to turn these off.

Reply:
I will change this to be the default.